src/middleware/middleware.js

```diff
@@ -31,4 +32,8 @@ export default function ({ query, route, $pinia }) {
   if (process.client && navStore.isReferredFromCc) {
     navStore.setIsReferredFromCc(false)
   }
+  if (process.server) {
+    const providerStore = useProviderStore($pinia)
+    await providerStore.nuxtServerInit()
```

Comment:
What is this for? :thinking:

Reply:
We were loading the array of providers with their stats on every server load by calling `nuxtServerInit` hook. This is the Pinia equivalent for it.
I guess we could just load a json file and update it regularly, similar to how we treat the i18n locale files.

Reply:
Ohhh, so this shares the store state between requests?

Reply:
Well, I tested it, and it doesn't  :( It calls the API on each page reload I make.

Reply:
Yeah, I wouldn't think that it would work this way.

If we need to share data between requests then we can do so with a singleton object in a module that gets initialized by the first request that hits it. We'd also need an accompanying accessible promise to act as an async lock for subsequent requests to await if they come in while the provider data is still being fetched.